pkg/antctl/raw/featuregates/command_test.go

```diff
@@ -0,0 +1,393 @@
+    server: https://localhost
+  name: fake-cluster
+contexts:
+- context:
+    cluster:  fake-cluster
+    user:  user-id
+  name:  fake-cluster
+current-context:  fake-cluster
+kind: Config`)
+
+	var err error
+	fakeKubeconfig, err := os.CreateTemp("", "fakeKubeconfig")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer os.Remove(fakeKubeconfig.Name())
+	fakeKubeconfig.Write(fakeConfigs)
+	kubeconfig := ""
+	Command.Flags().StringVarP(&kubeconfig, "kubeconfig", "k", fakeKubeconfig.Name(), "path of kubeconfig")
+	// Get kubeconfig with default config.
+	config, _, _, err := getConfigAndClients(Command)
+	assert.NoError(t, err)
+	assert.Equal(t, "https://localhost", config.Host)
+
+	// Get kubeconfig with new server option.
+	server := ""
+	Command.Flags().StringVarP(&server, "server", "", "http://192.168.1.10", "address and port of the API server")
+	newconfig, _, _, err := getConfigAndClients(Command)
+	assert.NoError(t, err)
+	assert.Equal(t, "http://192.168.1.10", newconfig.Host)
+}
+
+func TestGetRestClientByMode(t *testing.T) {
+	k8sClient := k8sfake.NewSimpleClientset(node1.DeepCopyObject())
+	tests := []struct {
+		name            string
+		expectedErr     string
+		antreaClientset antrea.Interface
+		mode            string
+	}{
+		{
+			name:        "failed to get rest client by agent mode",
+			expectedErr: "failed to create rest client: GroupVersion is required when initializing a RESTClient",
+			mode:        "agent",
+		},
+		{
+			name:            "get rest client by remote mode successfully",
+			antreaClientset: antreafakeclient.NewSimpleClientset(controllerInfo.DeepCopyObject()),
+			mode:            "remote",
+		},
+		{
+			name:            "failed to get rest client by remote mode",
+			expectedErr:     "failed to create rest client: error when creating controller client config: antreacontrollerinfos.crd.antrea.io \"antrea-controller\" not found",
+			antreaClientset: antreafakeclient.NewSimpleClientset(),
+			mode:            "remote",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := getRestClientByMode(clientConfig, k8sClient, tt.antreaClientset, tt.mode)
+			if tt.expectedErr == "" {
+				assert.NoError(t, err)
+			}
+			if err != nil {
+				assert.Equal(t, tt.expectedErr, err.Error())
```

Comment:
I think you can use `assert.EqualError` here

Reply:
Done